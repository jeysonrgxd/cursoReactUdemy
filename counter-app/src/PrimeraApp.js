// import React, { Fragment } from 'react'
import React from 'react'

// importamos una libreria que nos ayudara a las personas que usen nuestrpo componentes pasar los props adecuados
import PropTypes from 'prop-types'


// FC (functional components)
const PrimeraApp = ({ saludar, subtitulo }) =>{
   const saludo = "Jeyson"
   const valores = [1,2,3,4,5,6]
   const objeto = { 
      saludo
   }

   return (
      <>
         <h1>Hola Mundo {saludo}</h1>
         <p>{valores}</p>
         <p>{objeto.saludo}</p>
         <pre>{JSON.stringify(objeto,null,2)}</pre>
         <p>{saludar}</p>
         <p>Mi primera aplicacion</p>
         <p id="subtitle">{subtitulo}</p>
      </>
   );
}

// de esta forma nos aseguramos que los props de nuestro componente tiene que ser de un tipo de valor y que sea requerido osea que nos pasen ese valor
PrimeraApp.propTypes = {
   saludar : PropTypes.string.isRequired
}

// si utilizamos esta forma de definir por defecto el valor de nuestras props, tendremos el prop en nuestro componente, que adiferencia que si establecemos un valor por defecto en los parametro que recive la funcion
PrimeraApp.defaultProps = {
   subtitulo:'Soy un subtitulo'
}

export default PrimeraApp;