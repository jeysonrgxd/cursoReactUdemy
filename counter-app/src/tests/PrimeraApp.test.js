// imr y le das tab
import React from 'react'

import '@testing-library/jest-dom'

// import {render} from '@testing-library/react'

import PrimeraApp from '../PrimeraApp'

import { shallow } from 'enzyme'

describe('Pruebas en <PrimeraApp />', () => {

   // test('debe de mostrar el mensaje "Hola soy goku"', () => {
   //    const saludo = "Hola, soy Goku"

      // creamos el producto rederizado de mi componente
      // usamos la destructuracion para obtener la funcion que nos permitira evaluar
      // const { getByText } = render(<PrimeraApp saludar = {saludo}></PrimeraApp>)

      // aca evaluamos si en renderizado del componente existe el texto de la variable saludo
      // para eso usamos la funcion toBeInTheDocument()
      // expect(getByText(saludo)).toBeInTheDocument()
   // })

   test('debe de mostrar <PrimeraApp /> correctamente', () => {
      const saludo = 'Hola, soy Goku'
      // shallow parecido a render
      const wrapper = shallow(<PrimeraApp saludar = {saludo} />)
      expect( wrapper ).toMatchSnapshot()
   })

   test('debe de mostrar el subtitulo enviado por props', ()=>{
      const saludo = 'Hola, soy Goku'
      // shallow parecido a render
      const wrapper = shallow(<PrimeraApp saludar={saludo} />)
   } )

   test("Comparamos el texto del subitulo", ()=>{
      const saludo = 'Hola, soy Goku'
      const subTitulo = 'Soy un subtitulo'
      const wrapper = shallow(
      <PrimeraApp
          saludar={saludo}
         subtitulo = {subTitulo}
      />)

      // ahora gracias alalibreria de airbnb que ahora es de facebook podemos validar la informacion generada al renderizar nuestro componente
      const textoParrafo = wrapper.find('#subtitle').text()
      expect(textoParrafo).toBe(subTitulo);

   })

})