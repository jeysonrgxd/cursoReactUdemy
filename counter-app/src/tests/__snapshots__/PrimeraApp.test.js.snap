// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola Mundo 
    Jeyson
  </h1>
  <p>
    1
    2
    3
    4
    5
    6
  </p>
  <p>
    Jeyson
  </p>
  <pre>
    {
  "saludo": "Jeyson"
}
  </pre>
  <p>
    Hola, soy Goku
  </p>
  <p>
    Mi primera aplicacion
  </p>
  <p
    id="subtitle"
  >
    Soy un subtitulo
  </p>
</Fragment>
`;
